src/software/simulation/physics/physics_world.h

```diff
@@ -34,13 +40,131 @@ class PhysicsWorld
+    /**
+     * Returns the states and IDs of all yellow robots in the physics world.
+     *
+     * There is no guarantee as to what order the robots are returned in.
+     *
+     * @return the states and IDs of all yellow robots in the physics world
+     */
+    const std::vector<RobotStateWithId> getYellowRobotStates() const;
+
+    /**
+     * Returns the states and IDs of all blue robots in the physics world.
+     *
+     * There is no guarantee as to what order the robots are returned in.
+     *
+     * @return the states and IDs of all blue robots in the physics world
+     */
+    const std::vector<RobotStateWithId> getBlueRobotStates() const;
+
+    /**
+     * Returns the current timestamp of the physics world
+     *
+     * @return the current timestamp of the physics world
+     */
+    const Timestamp getTimestamp() const;
+
+    /**
+     * Sets the field in the physics world. The field must be valid. If an invalid
+     * field is provided the field is not updated.
+     *
+     * @param field the new field
+     */
+    void setField(const Field& field);
+
+    /**
+     * Sets the state of the ball in the physics world. No more than 1 ball may exist
+     * in the physics world at a time. If there is no ball in the physics world, a ball
+     * is added with the given state. If a ball already exists, it's state is set to the
+     * given state.
+     *
+     * @param ball_state The new ball state
+     */
+    void setBallState(const BallState& ball_state);
+
+    /**
+     * Removes the ball from the physics world
      */
-    World getWorld() const;
+    void removeBall();
 
     /**
-     * Advances the Box2D world physics by the given time step
+     * Adds robots to the yellow team with the given initial states. The robot IDs must
+     * not be duplicated and must not match the ID of any robot already on the yellow
+     * team.
+     *
+     * @throws runtime_error if any of the given robot ids are duplicated, or a
+     * yellow robot already exists with the ID
+     *
+     * @param robots the robots to add
+     */
+    void addYellowRobots(const std::vector<RobotStateWithId>& robots);
+
+    /**
+     * Adds robots to the yellow team with the given initial states. The robot IDs must
+     * not be duplicated and must not match the ID of any robot already on the yellow
+     * team.
+     *
+     * @throws runtime_error if any of the given robot ids are duplicated, or a
+     * yellow robot already exists with the ID
+     *
+     * @param robots the robots to add
+     */
+    void addBlueRobots(const std::vector<RobotStateWithId>& robots);
+
+    /**
+     * Returns the lowest available robot ID that is not already in use by a yellow robot.
+     * If no ID is available, returns std::nullopt.
+     *
+     * @return an available robot ID that is not already in use by a yellow robot.
+     * If no ID is available, returns std::nullopt
+     */
+    std::optional<unsigned int> getAvailableYellowRobotId() const;
```

Comment:
another possibility is to throw an exception if we run out of numbers

Reply:
lol fine.

Reply:
done, but used `LOG(FATAL)`